src/pextlib1.0/tracelib.c

```diff
@@ -614,10 +616,48 @@ static void dep_check(int sock, char *path) {
         answer(sock, "#");
     }
 
+    if (-1 == fs_cs) {
+        /*
+         * We need to initialize fs_cs.
+         * It's easier to do so when we actually have an existing file.
+         */
+
+        char *lowercase_path = malloc(strlen(path) + 1);
+        strlcpy(lowercase_path, path, strlen(path) + 1);
+
+        for (char *tmp_ptr = lowercase_path; *tmp_ptr; ++tmp_ptr) {
+            *tmp_ptr = tolower(*tmp_ptr);
+        }
+
+        struct stat path_stat, lowercase_path_stat;
+        if (-1 == lstat(path, &path_stat)) {
+            ui_error(interp, "unable to get information on file %s", path);
+            answer(sock, "#");
+        }
+
+        if (-1 == lstat(path, &lowercase_path_stat)) {
+            fs_cs = 1;
+        }
+        else {
+            /*
+             * Both files exists, but we must make sure they
+             * are really the same file.
+             */
+            if (path_stat.st_ino == lowercase_path_stat.st_ino) {
+                /* Truly case-insensitive! */
+                fs_cs = 0;
+            }
+            else {
+                /* Just similar-named, still a case-sensitive FS. */
+                fs_cs = 1;
+            }
+        }
+    }
```

Comment:
Yeah, could do. I found `getattrlist` while searching around and it might be good enough for `pextlib` and anything that only ever gets built and executed on OS X, but that's about it. Lowercasing the path and checking... which reminds me, we also have to check the uppercase version in case the original path happens to be lowercased in the first place (which is mostly the case!)

The best way to do this is probably by using a combination of `getattrlist` on Darwin and (either if it's not Darwin or `getattrlist` failed) a fallback that does the lowercase-uppercase dance otherwise.